Mage.Sets/src/mage/cards/t/ThreefoldSignal.java

```diff
@@ -0,0 +1,81 @@
+package mage.cards.t;
+
+import mage.MageObject;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.common.continuous.EachSpellYouCastHasReplicateEffect;
+import mage.abilities.effects.keyword.ScryEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.ModalDoubleFacesCardHalf;
+import mage.cards.SplitCardHalf;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.filter.FilterSpell;
+import mage.filter.predicate.Predicate;
+import mage.game.Game;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class ThreefoldSignal extends CardImpl {
```

Comment:
The way this card works in practice is also strange:
1. It asks you how many times you wish to pay replicate cost (say you do it twice)
2. Mana cost of the spell is shown and you are asked to pay (replicate cost is not visible)
3. *The first* replicate cost is shown. Once you pay it the next one appears.
4. This repeat until you've paid them all, but there's no way to know how many are left.

I don't know why all of the costs are not shown at once the way other additional costs are.

Reply:
I noticed that about how replicate is implemented, we need to refactor `OptionalAdditionalSourceCosts` the same way I did `AlternativeSourceCostsImpl` but I think we can do that outside this PR

Reply:
I'll add it to the TODO list at the top of the PR